node.go

```diff
@@ -357,6 +372,24 @@ func (n *Node) hashKey(b []byte) []byte {
 	return hasher.Sum(nil)
 }
 
+func (n *Node) nodePrototype(node schema.TypedPrototype) datamodel.NodePrototype {
+	if n.modeFilecoin {
```

Comment:
I don't think this function is necessary because the need to store using the representation should be consistent with or without `modeFilecoin`, the differences in the mode more subtle than that; if it's a bug for `modeFilecoin` then it's a bug for spec mode, so this function should just be replaced with a call to `.Representation()` shouldn't it?